### ISSUE TITLE
Unable to play YouTube playlist from specified track

### ISSUE THREAD

**Issue by xak2000** [TeamNewPipe/NewPipe#2510]
In last v0.17 there is no option to start playing YouTube playlist from specified track (in background or even in foreground).

Previously I used v0.16.2 and I can open YouTube playlist, long tap on any track in this playlist and start playing my YouTube playlist from this track in backgroud.

Now, when I do the same, the selected track also starts to play, but this time it is enqueued into local playlist alone, losing any relation to the YouTube playlist, from which it was started. And next track in the local queue is not my playlist's next track, but randomly suggested track.

So, it works like if I just enqueued a single track from my playlist, while previously it respected my playlist and played all tracks from it one by one (starting from the selected track). Actually, previously it enqueued all tracks from YouTube playlist (starting from currently selected track) into local queue. So they played in order one by one and looked like my YouTube playlist is playing (while actually it was still a local queue). It was very convinient.

Now the only way to do the same is to manually add all tracks from YouTube playlist into local queue (there is a button for it) and then go to the queue, scroll down to the specified track and tap on it to play from this position. Much more steps involved in the process.

Maybe it is just me, who mainly playing my youtube playlists from the middle. But it is really main mode how I use this application and it perfectly worked before v0.17.

**Comment by BartFarley** [comment]
Yes, the "play from here" menu item is gone in ver. 17.0
Please return it. One of your developers deemed it "useless",
so, he just... DELETED it. Wasn't useless to me.

**Comment by Stypox** [comment]
It was my fault, sorry :-(
I didn't consider it useless: I just didn't know that feature existed, so I accidentally changed the behaviour of "start playing in the background/popup". I will send a correcting pull request.

**Comment by Stypox** [comment]
Start playing on background/popup exists while start playing on main player does not. I didn't think that was useful, see [here (4th paragraph)](https://github.com/TeamNewPipe/NewPipe/pull/2368#issue-283397509) why.
Since this issue seems to be about starting background/popup from specified track, I submitted #2518 without thinking about the main player, sorry for the misunderstanding. I will not add back `start playing here on main player` feature, since then the menu would be too crowded and people would start complaining about it. :-/

An UI redesign is planned for the long press menu, that would include every feature you and others need without being too crowded. See #2354 #2383 #2198

**Comment by BartFarley** [comment]
Stypox,
Do me a favor. Install v0.16.2 on your device. Better yet, install ANY version prior to 0.17.0.
Then, open a channel, or a playlist. Long press any video you see. In the menu that appears,
you will see "start playing here". Every version prior to 0.17.0 had this option. It plays the list
of videos, full screen, starting at the selected video.  This is what NewPipe does. It plays lists 
of videos.  By removing this functionality, which you were not asked to remove, you have taken
away one of NewPipe's main purposes.
Back to v0.16.2 I go.

**Comment by Stypox** [comment]
I know which button you refer to. And yes, nobody asked me to remove that entry, but at the same time nobody complained up until now. In #2368 I explain why I did that and since everybody was ok with it the pr was merged.

I can see your point, and as I said earlier a redesign that suits everyone's needs is on its way. Again, the menu is already too crowded and people would complain about a bigger one, so a compromise has to be made. Also, it's only a waste of time to fine-tune a temporary feature that will be replaced. Since this is **in your opinion** a core feature of NewPipe, I will start working on the redesign and will ask for your feedback.

**Comment by gkeegan** [comment]
@BartFarley please try to not be rude over this.

**Comment by astamminger** [comment]
Hi @Stypox 

I understand what you say in #2368 but are you aware that starting the video from the details page will only enqueue a single video (i.e. the one you started)? From that perspective, deeming "start playing here" as useless isn't justified at all. I have to admit that I was using the "start playing here" option **a lot** when dealing with video playlists. For me that was the only way to **watch** (not talking about audio) the playlist with upcoming videos automatically enqueued. 

Don't get me wrong, I understand that this is a young project which changes a lot between releases which is okay an necessary, but I don't understand the motivation for removing a feature before any alternative is in place.

I hope this feature (or an alternative) will be available again in the near future (I'll downgrade to 0.16.2 for the time being)

Cheers and keep it going!

**Comment by Stypox** [comment]
@astamminger tell me what you think about the alternative in #2583

**Comment by astamminger** [comment]
@Stypox I think this is a very reasonable alternative which brings back the original behaviour with a much nicer GUI. I really like the redesign you proposed, very nice work 👍

One more thing that came to my mind: 
There's all this discussion going on about "Start from here". What I was asking myself: Instead of providing all these extra options, why not only use a single play button and then let the "Auto-enqueue next stream" option available from the settings decide whether we want to automatically play the next video in the playlist / search / whatever or if we just want to stop after playing the selected video?

What's your opinion on that? (May be I'm missing something :) )

**Comment by astamminger** [comment]
Sorry for the late reply @xak2000 !

> Obviously, it will be not usable at all if they required to go to settings each time to switch the default action of play buttons. 

But that's the current default, right? At least for the "simple" play button it'll always enqueue a next video unless you uncheck the auto-enqueue feature in the settings.
One would of course need to move that checkbox out of the settings to a more prominent position (similar to what the YouTube app is doing).

Nevertheless, I see your point and since there will always be a need for at least two playback buttons (play and play in background) I also think it is more coherent to have multiple playback buttons. But then IMHO one should get rid of the "Auto-enqueue next stream" option and use the "play from here" button everywhere to have content enqueued automatically.

**Comment by xak2000** [comment]
> But that's the current default, right?

Right. And this is why I don't use "just play this and only this track" function. To be honest, lately I only use "play in background from here" function 100% of the time.

**Comment by LukasThyWalls** [comment]
Hello, sorry for revamping.

I don't know if someone can explain me how works the alternative implemented to get rid the "Play from here" option to make the same behaviour.

Recently https://github.com/TeamNewPipe/NewPipe-legacy has been updated to the new UI elements in the last version, and i can't do the same anymore (or i don't know how to do it). I have looked the last NewPipe version and are the same UI elements.

I added a new playlist to NewPipe, with a bunch of songs for example, and i played it from the beginning. Sometime later, i closed the app at some song, but when i returned, i went to the playlist, selected that song i listened and used the "Play From here" option. With that i continued the playlist where i left, because it was following the same playlist but not from the beginning.

I know there is the "Background" and "Popup" play from here option, but i want to see them in fullscreen, normally, because i want to see the video. If i click a video, it only plays that video and forgets the playlist, and the only option is play it from the beginning and select the video in the playlist inside the player.

How can i do this now with the new UI elements?
